### ISSUE TITLE
secrets are written back to file with aws sm

### ISSUE THREAD

**Issue by ankitgupta-e** [kentik/ktranslate#574]
using the documentation here - https://docs.newrelic.com/docs/network-performance-monitoring/advanced/advanced-config/#external-config-files for AWS Secrets Manager -
default_v3:
user_name: aws.sm.SECRET_NAME_1
authentication_protocol: MD5
authentication_passphrase: aws.sm.SECRET_NAME_2
privacy_protocol: AES256
privacy_passphrase: aws.sm.SECRET_NAME_3

After the container is started, it writes back the secrets on the file system.

It shouldn't overwrite the file with sensitive information.

**Comment by araddas** [comment]
@ankitgupta-e you can write the secret in SM in a JSON format with all of the required configuration fields and reference it like this instead

```
default_v3: aws.sm.secret.name
```

**Comment by i3149** [comment]
Good catch -- `default_v3: aws.sm.secret.name` is the preferred way and the other approach should be deprecated. @Mesverrum , can you update the docs to reflect this?

**Comment by Mesverrum** [comment]
I've submitted the docs change to the content team, should go live in the next few days with the corrected syntax
